extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -643,6 +644,18 @@ class CornerstoneViewportService extends PubSubService implements IViewportServi
 
     return viewport.setStack(imageIds, initialImageIndexToUse).then(() => {
       viewport.setProperties({ ...properties });
+
+      const { customizationService } = this.servicesManager.services;
```

Comment:
This doesn't really belong here - this is adding specific behaviour for certain types of images in certain cases.  I would suggest adding it to the split/sort rules as part of generating the splitting criteria, and add it into the series level splitting for MG type series that have this issue.  Then, the hanging protocol rules can choose to pickup those values for the orientation/position.  That is a much more generic approach, and doesn't cause the viewport service to have deep knowledge about how to handle orientation changes.
You could also create a custom orientation/position metadata module as an alternative, and use that module in CS3D as the initial flip setup if it isn't currently set, but if so you need to change the viewport handling so that a viewport with flipHorizontal/flipVertical/initial rotation works correctly when you flip the viewport - that is, flipping the viewport should flip the images FROM the original location.

Reply:
In fact I think this should also be calculated in the hanging protocol service because the same orientation isn't always shown for hanging protocols, and those should still work correctly - that is, left on right or right on left should both be supported by the hanging protocol, and NOT the viewport.  Even if you create a default rule and add it into the default hanging protocol, I think doing it there is a better choice, although getting default values in the split/sort rules to apply at hte series level makes that easy to apply in the hanging protocol.

Reply:
What you should do is compute the desired default initial flip state in the display set object, and export it as a metadata instance and apply it here when it is set for the image id/display set uid.  It should be applied as the base data, and then rotation/flip etc on TOP of that applies relative to the base rotation etc.  That change could actually go into the CS3D library in the stack/volume viewports, or it can be done here as an integrated value.  That way, the image "flip" isn't shown by default unless the flip is changed relative to the base flip/rotate state.  That is a robust way to specify the values correctly.